ThunderTable/TableViewController.swift

```diff
@@ -136,26 +136,71 @@ open class TableViewController: UITableViewController {
         tableView.register(defaultNib, forCellReuseIdentifier: "Cell")
     }
 	
-	private var dynamicChangeObserver: NSObjectProtocol?
-	
-	public var shouldRedrawWithContentSizeChange = true
+    private var dynamicChangeObserver: NSObjectProtocol?
+    
+    private var accessibilityObservers: [Any] = []
+    
+    public var adjustsFontForContentSizeCategory: Bool = true
+    
+    /// A list of notification names that should cause the table view to redraw itself
+    public var accessibilityRedrawNotificationNames: [Notification.Name] = [
+        UIAccessibility.darkerSystemColorsStatusDidChangeNotification,
+        UIAccessibility.boldTextStatusDidChangeNotification,
+        UIAccessibility.grayscaleStatusDidChangeNotification,
+        UIAccessibility.invertColorsStatusDidChangeNotification,
+        UIAccessibility.reduceTransparencyStatusDidChangeNotification
+    ]
 	
 	open override func viewWillAppear(_ animated: Bool) {
 		
 		super.viewWillAppear(animated)
 		
-		dynamicChangeObserver = NotificationCenter.default.addObserver(forName: UIContentSizeCategory.didChangeNotification, object: self, queue: .main) { [weak self] (notification) in
-			guard let strongSelf = self, strongSelf.shouldRedrawWithContentSizeChange else { return }
-			strongSelf.tableView.reloadData()
+		dynamicChangeObserver = NotificationCenter.default.addObserver(forName: UIContentSizeCategory.didChangeNotification, object: nil, queue: .main) { [weak self] (notification) in
+			guard let strongSelf = self, strongSelf.adjustsFontForContentSizeCategory else { return }
+            strongSelf.reloadVisibleRowsWhilstMaintainingSelection()
+            strongSelf.accessibilitySettingsDidChange()
 		}
+        
+        // Notification names that it makes sense to redraw on
+        let accessibilityNotifications: [Notification.Name] = [
```

Comment:
Is this a duplication of `accessibilityRedrawNotificationNames` (just with a few more) - what's the reasoning behind this? :)

Reply:
See other (ThunderCloud) PR. This is so we don't have to have messy code and risk double-adding (or removing) these observers in the `didSet` for `accessibilityRedrawNotificationNames`. We'd also have to keep track of whether the VC is visible and use that in the `didSet` method. I figured it was easier to just listen to all notifications and then check in the callback as to whether the given notification name is present in `strongSelf.accessibilityRedrawNotificationNames`